src/main/java/org/openstreetmap/atlas/geography/atlas/raw/slicing/RawAtlasSlicer.java

```diff
@@ -320,44 +875,656 @@ protected boolean isInsideWorkingBound(final AtlasEntity entity)
+            // to a future edge
+            this.stagedPoints.remove(point.getIdentifier());
+            this.changes.add(FeatureChange.remove(CompletePoint.shallowFrom(point)));
+        }
+        else
+        {
+            final CompletePoint updatedPoint = this.stagedPoints.get(point.getIdentifier());
+            final SortedSet<String> countries = new TreeSet<>();
+            countries.addAll(
+                    Arrays.asList(getCountryBoundaryMap().getCountryCodeISO3(point.getLocation())
+                            .getIso3CountryCode().split(ISOCountryTag.COUNTRY_DELIMITER)));
+            updatedPoint.withAddedTag(ISOCountryTag.KEY,
+                    String.join(ISOCountryTag.COUNTRY_DELIMITER, countries));
+            if (countries.size() > 1)
+            {
+                updatedPoint.withAddedTag(SyntheticBoundaryNodeTag.KEY,
+                        SyntheticBoundaryNodeTag.EXISTING.toString());
+            }
+            if (!this.isInCountry.test(updatedPoint))
+            {
+                this.stagedPoints.remove(point.getIdentifier());
+                this.changes.add(FeatureChange.remove(updatedPoint, this.startingAtlas));
+            }
+        }
     }
 
-    private String getShardOrAtlasName(final Atlas atlas)
+    /**
+     * Take a polygon and find its slices, while also checking for validity and geometric
+     * consistency (i.e. only Polygonal results)
+     *
+     * @param polygon
+     *            The Polygon being sliced
+     * @param intersectingBoundaryPolygons
+     *            All intersecting country boundary polygons
+     * @return A SortedMap of country codes to the portions of the Polygon inside those country
+     *         boundary polygons
+     */
+    private SortedMap<String, Set<org.locationtech.jts.geom.Polygon>> slicePolygonGeometry(
+            final org.locationtech.jts.geom.Polygon polygon,
+            final Set<org.locationtech.jts.geom.Polygon> intersectingBoundaryPolygon)
     {
-        return atlas.metaData().getShardName().orElse(atlas.getName());
+        final Map<String, Set<Geometry>> currentResults = sliceGeometry(polygon,
+                intersectingBoundaryPolygon);
+
+        final SortedMap<String, Set<org.locationtech.jts.geom.Polygon>> results = new TreeMap<>();
+        for (final Map.Entry<String, Set<Geometry>> entry : currentResults.entrySet())
+        {
+            final String countryCode = entry.getKey();
+            final Set<org.locationtech.jts.geom.Polygon> slicedPolygons = new HashSet<>();
+            entry.getValue().forEach(geometry ->
+            {
+                if (geometry instanceof org.locationtech.jts.geom.Polygon)
+                {
+                    if (((org.locationtech.jts.geom.Polygon) geometry).getNumInteriorRing() > 0)
+                    {
+                        logger.error("Discarding multipolygon slicing result");
```

Comment:
Is this expected to be common?  If not, should there be more description of the geometry to troubleshoot?

Reply:
This case is very rare, but it's important to discard. I tried to design a unit case around this but struggled to get it to replicate properly, basically it happens when a closed Line that will become an Area has an inner hole introduced by a boundary. In that case, our Area representation of a simple Polygon with no holes won't work, and we're forced to either discard the slicing result or include it without the hole (which seems much worse!).

Reply:
That seems like a reasonable outcome.  The open question is more -- can the geometry or at least some location of the feature get logged to help rare case future troubleshooting?  Most other error logs here give more hints about the the geometry or feature in trouble.

Reply:
in light of this discussion, I've changed the behavior-- now if this happens, we throw an exception, catch it upstream a bit, and slice the feature linearly instead of polygonally. this should resolve any concerns about bad geometry for this kind of issue.

Reply:
Oh nice!  Thank you @adahn6